### PR TITLE
Turn all components into classes

### PR DESCRIPTION
**Pull request by unindented**
Function components don't have a display name, which was making things harder to debug for people.

### CODE REVIEW

**Comment by unindented** [comment]
@bryan-gislason please review.

File: src/components/Button/index.tsx [hunk 1]
```diff
@@ -23,61 +23,61 @@ export interface BaseButtonProps extends BaseComponentProps {
...
+  onMouseLeave?: (event: any) => void;
 }
 
 export interface RegularButtonProps extends BaseButtonProps {
   /**
-   * Whether this button should be disabled or not. Defaults to false.
+   * Disabled buttons cannot have a link href.
    */
-  disabled?: boolean;
+  href?: void;
```
Comment (bryan-gislason) [inline src/components/Button/index.tsx]: why is this void and not a string?

File: src/components/MediaObject/__snapshots__/media-object.test.tsx.snap [hunk 2]
```diff
@@ -4,92 +4,92 @@ exports[`<MediaObject /> with additional className matches its snapshot 1`] = `
 <div
   className="y-media-object y-media-object__size-medium TEST_CLASSNAME"
 >
-  <Component
+  <FixedGridRow
```
Comment (bryan-gislason) [inline src/components/MediaObject/__snapshots__/media-object.test.tsx.snap]: Much better!

File: src/components/Button/index.tsx (hunk 1 above)
Comment (unindented) [reply src/components/Button/index.tsx]: We discussed in person. We think they're there to disambiguate the union.

**Pull request merged**